.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,22 @@
+# https://pre-commit.com
+# This GitHub Action assumes that the repo contains a valid .pre-commit-config.yaml file.
+# Using pre-commit.ci is even better than using GitHub Actions for pre-commit.
+name: pre-commit
+on:
+  pull_request:
+    branches: [develop]
+  push:
+    branches: [develop]
+  workflow_dispatch:
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.x
+      - run: pip install pre-commit
+      - run: pre-commit --version
+      - run: pre-commit install
+      - run: pre-commit run --all-files
```

Comment:
@cclauss  I am a little bit confused.
All this is good but now we'll have `codespell` running in `macOS-tests.yml` and in `pre-commit.yml` if we open a PR against `develop` branch. And when `swiftlint` will be enabled in `pre-commit-config.yaml` (and I think it should be enabled now with the update you made to `disabled_rules`) we will have both `swiftlint` and `codespell` running twice.

Reply:
What was again the purpose of this workflow?

Reply:
Enable SwiftLint in GitHub Actions but disable the failing tests that cannot be auto-fixed so that we do not backslide on the tests that currently pass.